uploadNeuroDB/NeuroDB/MRIProcessingUtility.pm

```diff
@@ -1770,72 +1804,96 @@ or a phantom
-        $CandMismatchError= 'PSCID does not exist';
+        $message = "\n\n=> Could not find candidate with PSCID=$pscID in database\n";
+        $this->writeErrorLog($message, $NeuroDB::ExitCodes::INSERT_FAILURE);
+        $this->spool($message, 'Y', $upload_id, $notify_notsummary);
+
+        $CandMismatchError = 'PSCID does not exist';
+
         return $CandMismatchError;
     } else {
         # Check that the PSCID and CandID refers to the same candidate
         my $rowref = $sth->fetchrow_hashref;
-        unless ($rowref->{'CandID'} == $subjectIDsref->{'CandID'}) {
-            $CandMismatchError = 'PSCID and CandID of the image mismatch\n';
-            print LOG "\n=> $CandMismatchError";
+        unless ($rowref->{'CandID'} == $candID) {
+            $CandMismatchError = "PSCID and CandID of the image mismatch\n";
+            $this->writeErrorLog($CandMismatchError, $NeuroDB::ExitCodes::INSERT_FAILURE);
+            $this->spool($CandMismatchError, 'Y', $upload_id, $notify_notsummary);
+
             return $CandMismatchError;
         }
     }
 
-    ############################################################
-    ################ No Checking if the subject is Phantom #####
-    ############################################################
-    if ($subjectIDsref->{'isPhantom'}) {
-        # CandID/PSCID errors don't apply to phantoms, so we don't
-        # want to trigger
-        # the check which aborts the insertion
-        $CandMismatchError = undef;
-        return $CandMismatchError;
-    }
 
     ############################################################
-    ################ Check if visitLabel exists ################
+    ################ Check if visit label is valid##############
     ############################################################
-
     $query = "SELECT Visit_label FROM Visit_Windows WHERE BINARY Visit_label=?";
-    $sth =  ${$this->{'dbhr'}}->prepare($query);
-    $sth->execute($subjectIDsref->{'visitLabel'});
+    $sth   = ${$this->{'dbhr'}}->prepare($query);
+    $sth->execute($visit_label);
+
     if (($sth->rows == 0) && (!$subjectIDsref->{'createVisitLabel'})) {
-        print LOG "\n=> No Visit label\n";
-        $CandMismatchError= 'Visit label does not exist';
+        $message = "\n=> No Visit label\n";
+        $this->writeErrorLog($message, $NeuroDB::ExitCodes::INSERT_FAILURE);
+        $this->spool($message, 'Y', $upload_id, $notify_notsummary);
+
+        $CandMismatchError = "Visit label $visit_label does not exist in Visit_Windows";
+
         return $CandMismatchError;
     } elsif (($sth->rows == 0) && ($subjectIDsref->{'createVisitLabel'})) {
-        print LOG "\n=> Will create visit label $subjectIDsref->{'visitLabel'}\n";
+        $message = "\n=> Will create visit label $visit_label in Visit_Windows\n";
+        $this->spool($message, 'Y', $upload_id, $notify_notsummary);
     } 
 
-   return $CandMismatchError;
+    return $CandMismatchError;
```

Comment:
So if all goes well, `$CandMismatchError` remains uninitialized and the method will return `undef`, right? If that's the case, I would state so explicitly with `return undef`.

Reply:
perfect! Done!